ledger/controllers/grants.js

```diff
@@ -547,16 +650,31 @@ async function captchaCheck (debug, runtime, request, promotion, wallet) {
   }
 }
 
-const grantsUploadSchema = {
-  grants: Joi.array().min(0).items(
-    Joi.string().required().description('the jws encoded grant')
-  ).description('grants for bulk upload'),
-  promotions: Joi.array().min(0).items(Joi.object().keys({
-    promotionId: Joi.string().required().description('the promotion-identifier'),
-    priority: Joi.number().integer().min(0).required().description('the promotion priority (lower is better)'),
-    active: Joi.boolean().optional().default(true).description('the promotion status')
-  }).unknown(true).description('promotions for bulk upload'))
-}
+const promotionIdValidator = Joi.string().required().description('the promotion-identifier')
+const priorityValidator = Joi.number().integer().min(0).required().description('the promotion priority (lower is better)')
+const activeValidator = Joi.boolean().optional().default(true).description('the promotion status')
+const protocolVersionValidator = Joi.number().integer().min(2).required().description('the protocol version that the promotion will follow')
+const promotionValidator = Joi.object().keys({
+  promotionId: promotionIdValidator,
+  priority: priorityValidator,
+  active: activeValidator
+}).unknown(true).description('promotions for bulk upload')
+const promotionsValidator = Joi.array().min(0).items(promotionValidator)
+const grantValidator = Joi.string().required().description('the jws encoded grant')
+const grantsValidator = Joi.array().min(0).items(grantValidator).description('grants for bulk upload')
+const grantsUploadSchema = Joi.object().keys({
+  grants: grantsValidator,
+  promotions: promotionsValidator
+}).required().description('data for bulk upload')
+const promotionV4Validator = promotionValidator.keys({
+  protocolVersion: protocolVersionValidator,
+  minimumReconcileTimestamp: Joi.number().optional().description('time when the promotion can be reconciled')
+})
+const promotionsV4Validator = Joi.array().min(0).items(promotionV4Validator)
+const grantsUploadV4Validator = grantsUploadSchema.keys({
+  grants: grantsValidator,
+  promotions: promotionsV4Validator
+})
```

Comment:
it looks like `grantsUploadV4Validator` and `grantsUploadSchema` are pretty similar.  is there difference in your mind between a validator and a schema? would it be possible to consolidate to using just one of those terms?

Reply:
i think this was just because one existed before, so the suffix was just to differentiate between old and new. i suppose i can just add old to it to mark deprication